cpp/src/parquet/encoding-benchmark.cc

```diff
@@ -163,4 +172,132 @@ static void BM_DictDecodingInt64_literals(benchmark::State& state) {
 
 BENCHMARK(BM_DictDecodingInt64_literals)->Range(1024, 65536);
 
+std::shared_ptr<::arrow::Array> MakeRandomStringsWithRepeats(size_t num_unique,
+                                                             size_t num_values) {
+  const int64_t min_length = 2;
+  const int64_t max_length = 10;
+  std::vector<uint8_t> buffer(max_length);
+  std::vector<std::string> dictionary(num_unique);
+
+  uint32_t seed = 0;
+  std::default_random_engine gen(seed);
+  std::uniform_int_distribution<int64_t> length_dist(min_length, max_length);
+
+  std::generate(dictionary.begin(), dictionary.end(), [&] {
+    auto length = length_dist(gen);
+    ::arrow::random_ascii(length, seed++, buffer.data());
+    return std::string(buffer.begin(), buffer.begin() + length);
+  });
+
+  std::uniform_int_distribution<int64_t> indices_dist(0, num_unique - 1);
+  ::arrow::StringBuilder builder;
```

Comment:
Since you know the number of values (`num_values`) that will be `Apppend`ed to the `arrow::StringBuilder`, it might be a good idea to call `builder.Reserve(num_values)` explicitly here.

Reply:
Good catch!